### PR TITLE
SnarkyJS ECDSA

### PR DESCRIPTION
**Pull request by Trivo25**
follow up for #9, for ECDSA

### CODE REVIEW

File: 0005-snarkyjs-ecdsa-sha.md [hunk 1]
```diff
@@ -148,31 +148,125 @@ Overall, exposing new gadgets and gates follow a strict pattern that has been us
...
+- `msg_hash` $\in Fn$ (`bytes_to_foreign_field_element` gadget)
+
+Each of these preconditions requires expensive checks so its important to choose wisely when and how to check these inputs. Its important to provide a safe API as well as giving experienced developers enough space to optimize their applications by avoiding double constraining of inputs.
+
+By building on top of the [`ForeignField`](https://github.com/o1-labs/snarkyjs/pull/985) implementation, we can leverage the modular approach taken by it to implement an ECDSA API which provides developers with a powerful modular interface. This approach allows us to potentially enable multiple versions of ECDSA with different curves, within the same API. We will also implement a `ForeignCurve` API which relies on the non-native elliptic curve primitives and gadgets introduce in the original [ECDAS PR](https://github.com/MinaProtocol/mina/pull/13279) so developers can not only profit from a modular ECDSA API, but can also access non-native elliptic curves directly.
+
+Similar to [`ForeignField`](https://github.com/o1-labs/snarkyjs/pull/985), the API of `ForeignCurve` will make use of a modular approach.
```
Comment (Trivo25) [inline 0005-snarkyjs-ecdsa-sha.md:191]: this is all @mitschabaude - I just compiled it into the RFC. Thank you very much!

File: 0005-snarkyjs-ecdsa-sha.md [hunk 2]
```diff
@@ -148,31 +148,158 @@ Overall, exposing new gadgets and gates follow a strict pattern that has been us
...
 This includes testing the out-of- and in-snark variants using our testing framework, as well as adding a SHA3 and Keccak regression test. The regression tests will also include a set of predetermined digests to make sure that the algorithm doesn't unexpectedly change over time (similar to the tests implemented for the OCaml gadget). We will include a range of edge cases in the tests (e.g. empty input, zero, etc).
 
 In addition to that, we should provide a dedicated integration test that handles SHA3/Keccak hashing within a smart contract (proving enabled). This will allow us to not only provide developers with an example, but also ensure that SHA3 and Keccak proofs can be generated.
 
+### ECDSA
+
+ECDSA testing will follow a similar approach to SHA3 and Keccak. We will utilize the regression test framework in SnarkyJS to make sure no unintended changes are introduced and backwards compatibility is guaranteed. Similar to the original ECDSA gadget tests, we will also test a set of pre-defined ECDSA signatures (e.g. Ethereum transaction signatures).
```
Comment (bkase) [inline 0005-snarkyjs-ecdsa-sha.md:287]: @shimkiv look good to you?

File: 0005-snarkyjs-ecdsa-sha.md [hunk 3]
```diff
@@ -148,31 +148,158 @@ Overall, exposing new gadgets and gates follow a strict pattern that has been us
...
 ## ECDSA
 
-**TODO** this will be a seperate PR stacked on top of this one
+Similar to SHA3 and Keccak, the gadget for ECDSA has been implemented by the crypto team and is available through exposing it in the bindings layer. However, designing and implementing a safe API for ECDSA is not as straight forward as it is for SHA3 and Keccak.
 
-**Evergreen, wide-sweeping Details**
+The main verification step of ECDSA is implemented via a `verify` gadget in OCaml.
 
-**Ephemeral details**
+```ocaml
```
Comment (bkase) [inline 0005-snarkyjs-ecdsa-sha.md:155]: All this information about the OCaml interface is great but should be in a comment in the OCaml implementation and linked to from this doc!
